extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -826,8 +798,500 @@ private Table readTableInternal(
+     * list of data files that were written. Users can use this list to create a transaction/snapshot if needed.
+     *
+     * @param tableIdentifier The identifier for the Iceberg table to write to.
+     * @param dhTable The Deephaven table containing the data to be written.
+     * @param instructions The instructions for customizations while writing, or null to use default instructions.
+     *
+     * @return A list of {@link DataFile} objects representing the written data files.
+     */
+    public List<DataFile> write(
+            @NotNull final TableIdentifier tableIdentifier,
+            @NotNull final Table dhTable,
+            @Nullable final IcebergWriteInstructions instructions) {
+        return write(tableIdentifier, new Table[] {dhTable}, instructions);
+    }
+
+    /**
+     * Writes data from Deephaven tables to an Iceberg table without creating a new snapshot. This method returns a list
+     * of data files that were written. Users can use this list to create a transaction/snapshot if needed. All tables
+     * should have the same definition, else a table definition should be provided in the instructions.
+     *
+     * @param tableIdentifier The identifier string for the Iceberg table to write to.
+     * @param dhTables The Deephaven tables containing the data to be written.
+     * @param instructions The instructions for customizations while writing, or null to use default instructions.
+     *
+     * @return A list of {@link DataFile} objects representing the written data files.
+     */
+    public List<DataFile> write(
+            @NotNull final String tableIdentifier,
+            @NotNull final Table[] dhTables,
+            @Nullable final IcebergWriteInstructions instructions) {
+        return write(TableIdentifier.parse(tableIdentifier), dhTables, instructions);
+    }
+
+    /**
+     * Writes data from Deephaven tables to an Iceberg table without creating a new snapshot. This method returns a list
+     * of data files that were written. Users can use this list to create a transaction/snapshot if needed. All tables
+     * should have the same definition, else a table definition should be provided in the instructions.
+     *
+     * @param tableIdentifier The identifier for the Iceberg table to write to.
+     * @param dhTables The Deephaven tables containing the data to be written.
+     * @param instructions The instructions for customizations while writing, or null to use default instructions.
+     *
+     * @return A list of {@link DataFile} objects representing the written data files.
+     */
+    public List<DataFile> write(
+            @NotNull final TableIdentifier tableIdentifier,
+            @NotNull final Table[] dhTables,
+            @Nullable final IcebergWriteInstructions instructions) {
+        return writeImpl(tableIdentifier, dhTables, instructions, false, false);
+    }
```

Comment:
Ditto; I wonder if we'd prefer a different method name to differentiate these from their transactional counterparts.

Reply:
I have renamed `write` to `writeDataFiles`.
Some other alternates can be 
- `writeDirect`
- `writeNoSnapshot`
- `writeRaw`

Is there something you prefer?